src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.cs

```diff
@@ -90,9 +90,24 @@ public MBuild.Project Project
                 {
                     return Context.ProjectCollection.LoadProject(FileInfo.FullName);
                 }
-                catch (InvalidProjectFileException ex)
+                catch (InvalidProjectFileException)
                 {
-                    throw new UpgradeException(LocalizedStrings.InvalidProjectError, ex);
+                    // TODO Delete obj file retry - a change in platform version may cause load to fail for UWP/WinUI apps
+                    var fileDirectory = FileInfo.Directory;
+                    var dirToDelete = fileDirectory?.EnumerateDirectories().Where(directory => directory.Name == "obj").FirstOrDefault();
+                    if (dirToDelete != null)
+                    {
+                        dirToDelete.Delete(true);
+                    }
+
+                    try
+                    {
+                        return Context.ProjectCollection.LoadProject(FileInfo.FullName);
```

Comment:
if `dirToDelete is null`, is there a point in re-trying this or should we just throw the exception?